src/main/java/controllers/GameSelector.java

```diff
@@ -58,26 +58,49 @@ public void run() {
 
             int sel = this.selectorInput.getUserSelection();
 
-            List<String> usernames = new ArrayList<>();
-
-            String username = this.selectorInput.getUsername();
+            if (sel == 0) {
+                return;
+            }
 
-            while(!username.equalsIgnoreCase("done")) {
-                usernames.add(username);
-                username = this.selectorInput.getUsername();
+            while (!checkValidity(sel)) {
+                this.selectorOutput.sendOutput("Invalid menu selection.\n");
+                sel = this.selectorInput.getUserSelection();
             }
 
+            String gameString = this.games[sel - 1];
+
+            List<String> usernames = new ArrayList<>();
+
             UserManager userManager = new UserManager();
 
-            // check for exit case
-            if (sel == 0) {
-                return;
+            int maxPlayers = GameTemplate.GetMaxPlayers(gameString);
+
+            System.out.println("Input up to " + maxPlayers + " usernames for " +
+                    "players playing the game. Enter 'done' to finish.");
+
+            String username = this.selectorInput.getUsername();
+
+            while (true) {
+                if (username.equalsIgnoreCase("done")) {
+                    System.out.println("\nPlease enter at least one username!\n");
+                    username = this.selectorInput.getUsername();
+                } else {
+                    break;
+                }
             }
 
-            while (!handleUserSelection(sel, usernames, userManager)) {
-                this.selectorOutput.sendOutput("Invalid menu selection.\n");
-                sel = this.selectorInput.getUserSelection();
+            while (!username.equalsIgnoreCase("done") && maxPlayers != 0) {
+                usernames.add(username);
+                try {
+                    userManager.addUser(username);
+                } catch (UserManager.UserAlreadyExistsException e) {
+                    e.printStackTrace();
+                }
```

Comment:
We shouldn't print the stack trace if the user already exists. It's normal behaviour to have the same username be used.